docs/adrs/0074-move-history-details-refactor.md

```diff
@@ -0,0 +1,112 @@
+
+**🔒 User Story:** *[MB-11214](https://dp3.atlassian.net/browse/MB-12606)*
+
+*Currently, move history templates are matched based on action, event name, and table name. This expects that each action, event and table combination. However there are some move history events that share the same values for all three and there isn't a way to specify which template to use. The event will then match to whichever template shows up first in the index file.*
+
+## Proposal: Clean up the current Template Manager
+
+To be able to handle changed values. We will first need to refactor the existing [Template Manager](https://github.com/transcom/mymove/blob/master/src/constants/MoveHistory/TemplateManager.js). We will need the template manager to be able to three things:
+
+1. [Match templates to the an event by action, event,  and table name.](#1-match-templates-to-the-an-event-by-action-event-and-table-name)
+2. [Match template to an event using the change values.](#2-match-template-to-an-event-using-the-change-values)
+3. [Determine if an event, needs to matched using changed values in addition matching with action, event, and table name.](#3-determine-if-an-event-needs-to-matched-using-changed-values-in-addition-matching-with-action-event-and-table-name)
+
+
+### 1. Match templates to the an event by action, event, and table name.
+The current template manager already does this. We can make this its own function that can be reused.
+```js
+   //before
+    return (
+      propertiesMatch(eventType.action, other?.action) &&
+      propertiesMatch(eventType.eventName, other?.eventName) &&
+      propertiesMatch(eventType.tableName, other?.tableName)
+    );
+
+    // after
+    function matchByDefaults(template, entry) {
+    return (
+      propertiesMatch(template.action, entry?.action) &&
+      propertiesMatch(template.eventName, entry?.eventName) &&
+      propertiesMatch(template.tableName, entry?.tableName)
+    );
+  }
+  ;
+
+```
+
+### 2. Match template to an event using the change values.
+To match the history record entry to the correct template, we will take the history record entry's changed value object and compare the keys of that object to the expected keys for the template.
+```js
+  function matchByChangedValues(template, entry) {
+    const entryChangedValues = Object.keys(entry.changedValues);
+    const templateChangedValues = template.getChangedValues();
+    return (
+      templateChangedValues.length === entryChangedValues.length &&
+      templateChangedValues.every((key) => entryChangedValues.indexOf(key))
+    );
+  }
+```
+### 3. Determine if an event needs to be matched using changed values in addition matching with action, event, and table name.
+
+We will only match a template using `changedValues` if a move template includes a list of changed values. Otherwise, we will only use action, event and table name to match a template. With this check, existing move templates will not need to be updated to list their `changedValues.
```

Comment:
```suggestion
We will only match a template using `changedValues` if a move template includes a list of changed values. Otherwise, we will only use action, event and table name to match a template. With this check, existing move templates will not need to be updated to list their `changedValues`.
```